src/bun.js/api/bun/socket.zig

```diff
@@ -1495,6 +1497,7 @@ fn NewSocket(comptime ssl: bool) type {
         pub fn onClose(this: *This, socket: Socket, err: c_int, _: ?*anyopaque) void {
             JSC.markBinding(@src());
             log("onClose", .{});
+            if (this.detached) return;
```

Comment:
shouldn't we still call the close callback here? `detached` refers to the TCP socket being connected, not the handlers

Reply:
i.e. if you call `.close`, it should run the `close` callback

Reply:
Yeah you are right let me remove this